counterparty-core/counterpartycore/lib/mempool.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import time
+
+from counterpartycore.lib import blocks, config, exceptions, ledger
+from counterpartycore.lib.kickstart import blocks_parser
+
+logger = logging.getLogger(config.LOGGER_NAME)
+
+
+def parse_mempool_transactions(db, raw_tx_list):
+    logger.debug("Parsing mempool block")
+    now = time.time()
+    transaction_events = []
+    cursor = db.cursor()
+    try:
+        with db:
+            # insert fake block
+            cursor.execute(
+                """INSERT INTO blocks(
```

Comment:
Is there a way to do the mempool without having this virtual block? It seems to me that was a bad way to do it originally. (My fault!)

Reply:
nope.. because of the foreign key constraints and the block_index field present in all tables we are obliged to have a corresponding block when we parse a mempool transaction.. it is also the most practical way to parse several transactions of the mempool at a time. I modified the "Parsing mempool block" log with "Parsing {len(raw_tx_list)} raw transactions from mempool...".
Of course this block, like all the mempool txs are never really recorded in the database.. we only retrieve the events and cancel the SQL transaction.